fix: Resolve login issues in frontend and backend

Bright-04 opened the pull request:
Key updates: The login process is now smoother with logical navigation (or redirection, idk). Plus, there's a "Remember Me" option now.

### Backend Changes

**Authentication Improvements:**
* I've made it so that the login information (JWT tokens) is stored in a safer type of cookie that only the server can access. This makes things more secure. I also changed what information is sent back after logging in – it no longer includes the login token. I've also made the error messages more helpful, especially when I'm working on the app (`auth.controller.js`).

**Code Cleanup:**
* I've removed an unused piece of code related to `Invoice` in `index.js` to keep things cleaner.

### Frontend Changes

**Routing and Navigation:**
* I've added secure routes using a new `ProtectedRoute` component. This means some pages can only be accessed if you're logged in (`App.jsx`, `ProtectedRoute.jsx`).

**Authentication Flow:**
* I've updated how the app sends requests so that it automatically includes the login cookies. I also removed the part where I was manually adding the login token to each request (`axios.js`).
* I've simplified the `loginApi` function. Now, it checks your login details, shows helpful messages, and only returns your user information. The login token is now handled by the cookies (`authApi.js`).

Bright-04 committed:
Enhance authentication flow with improved error handling and logging

- Added detailed logging for login attempts and results in auth.controller.js
- Implemented input validation for login credentials in authApi.js
- Updated login thunk in authSlice.js to handle errors more gracefully
- Refactored Login component to include form submission handling and loading state
- Created a script to hash passwords for secure storage

Bright-04 committed:
feat: Remove unused package and lock files

Remove unused package and lock files

HLoc26 committed:
Merge branch 'develop' into OOSE-91-Fully-implement-Login-function

Bright-04 committed:
refactor: standardize constructor formatting across multiple classes

HLoc26 reviewed (changes requested):
have some duplication that needs to be removed

Bright-04 committed:
Update API/src/server.js

Co-authored-by: Đặng Hữu Lộc <dhl26052004@gmail.com>

HLoc26 committed:
Merge branch 'develop' into OOSE-91-Fully-implement-Login-function

HLoc26 committed:
Merge branch 'develop' of https://github.com/HLoc26/VinaBooking into OOSE-91-Fully-implement-Login-function

Bright-04 committed:
feat: implement protected routes and user authentication flow

Bright-04 committed:
feat: implement JWT authentication with HTTP-only cookies and remove localStorage usage

No more XSS attacks. Thank you!

Bright-04 committed:
refactor: improve logging in authentication flow by removing sensitive data from logs

Console can no longer expose sensitive information...

Bright-04 committed:
refactor: enhance login form by adding loading state to input fields

You won't be able to accidentally click the login button too many times now.

Bright-04 commented:
### Backend Changes
* Updated the `login` method in `auth.controller.js` to set JWT as an HTTP-only cookie, remove the JWT from the response payload, and include detailed error handling for server issues.
* Commented out the unused `Invoice` export in `index.js` to prevent unnecessary imports or crashing application. (I bet you forgot this thing)

### Frontend Changes
* Introduced protected routes in `App.jsx` using a new `ProtectedRoute` component that redirects unauthenticated users to the login page. 
* Updated `NavBar.jsx` to dynamically display navigation options based on the user's authentication state, and added a logout functionality that clears user state and redirects to the homepage. 
* Refactored the `loginApi` function in `authApi.js` to validate credentials, log minimal information, and return only the user object (JWT is handled via cookies).

Bright-04 commented:
## Feature: "Remember Me"
**Backend Changes:**
* The authentication service has been updated to handle different token expiration durations. This is now determined by the "Remember Me" preference.
*  The login function in the backend now accepts a `rememberMe` parameter from the frontend. This parameter dictates how long the user's session should last.
* The settings for the authentication cookie in the backend have been modified to dynamically adjust the expiration time based on the token's validity period.

**Frontend Changes:**
* **"Remember Me" Checkbox:** A "Remember Me" checkbox has been added to the login form, allowing users to indicate their preference.
* The frontend API call for login now includes the value of the "Remember Me" checkbox, sending this preference to the backend.
* The frontend's state management (using Redux) has been updated to store and manage the "Remember Me" state.

## So, basically...
1.  When a user logs in, they will see a "Remember Me" checkbox.
2.  If the user **checks** the "Remember Me" box:
    * The backend will generate an authentication token that is valid for **30 days**.
    * The HTTP-only cookie storing this token will also be set to expire in **30 days**.
    * This means the user will remain logged in for 30 days on the same browser without needing to log in again.
3.  If the user **does not check** the "Remember Me" box (this is the default behavior):
    * The authentication token will have its default expiration time of **1 day**.
    * The cookie will also expire after **1 day**.
4. Please don't hit me.

Bright-04 committed:
feat: add "Remember Me" functionality to login process and update related components

I literally forgot commiting this ...

HLoc26 commented:
> * Commented out the unused `Invoice` export in `index.js` to prevent unnecessary imports or crashing application. (I bet you forgot this thing)

yup, fixed it in my latest PR... thanks for pointing that out

Bright-04 committed:
refactor: update login response to exclude JWT and improve payload structure

Bright-04 commented:
# Why I decided to switch from localStorage JWT to HTTP-only Cookies

> "My decision to switch from manual JWT to HTTP cookies was prompted by a frustrating experience with the FHQx site. Their implementation repeatedly caused login-logout loops, which I initially suspected was due to our failure to manually log out combined with their server's (presumed) lack of session expiration. At least, that was my thought."

## Concerns w/ localStorage

1. **XSS Vulnerability**
```javascript
// Current approach (vulnerable)
localStorage.setItem("jwt", data.jwt); // Anyone can access via JavaScript
```
- Any JavaScript code (including malicious scripts) can access localStorage
- Attackers can steal tokens through cross-site scripting (XSS) attacks

2. **No Built-in Security/Expiration Features**
- localStorage tokens stay forever until manually removed
- The token might be expired on the server, but still present in localStorage
- This creates a mismatch between client and server states
- No security flags or domain restrictions

## Benefits of HTTP-only Cookies

1. **Enhanced Security**
```javascript
// Server-side implementation
res.cookie("jwt", token, {
    httpOnly: true,    // Cannot be accessed by JavaScript
    secure: true,      // HTTPS only
    sameSite: "lax",   // CSRF protection
    maxAge: 24 * 60 * 60 * 1000 // 1 day expiration
});
```

2. **Simpler Frontend Code**
```javascript
// Before: Manual token management
localStorage.setItem("jwt", data.jwt);
localStorage.removeItem("jwt");

// After: Automatic handling
// No token management code needed in frontend. Ever.
// I am trying to keep it simple, Hloc...
```

3. **Better Authentication Flow**
- Automatic cookie transmission with requests
- Browser-level security features
- Server-controlled token lifecycle
